lib/ts/recipe/webauthn/index.ts

```diff
@@ -0,0 +1,408 @@
+import { getRecoverAccountLink } from "./utils";
+import { getRequestFromUserContext, getUser } from "../..";
+import { getUserContext } from "../../utils";
+import { SessionContainerInterface } from "../session/types";
+import { User } from "../../types";
+import {
+    DEFAULT_REGISTER_OPTIONS_REQUIRE_RESIDENT_KEY,
+    DEFAULT_REGISTER_OPTIONS_RESIDENT_KEY,
+    DEFAULT_REGISTER_OPTIONS_SUPPORTED_ALGORITHM_IDS,
+    DEFAULT_REGISTER_OPTIONS_USER_VERIFICATION,
+    DEFAULT_SIGNIN_OPTIONS_USER_VERIFICATION,
+} from "./constants";
+
+export default class Wrapper {
+    static init = Recipe.init;
+
+    static Error = SuperTokensError;
+
+    static async registerOptions(
+        email: string | undefined,
+        recoverAccountToken: string | undefined,
+        relyingPartyId: string,
+        relyingPartyName: string,
+        origin: string,
+        timeout: number,
+        attestation: "none" | "indirect" | "direct" | "enterprise" = "none",
+        tenantId: string,
+        userContext: Record<string, any>
+    ): Promise<
+        | {
+              status: "OK";
+              webauthnGeneratedOptionsId: string;
+              rp: {
+                  id: string;
+                  name: string;
+              };
+              user: {
+                  id: string;
+                  name: string;
+                  displayName: string;
+              };
+              challenge: string;
+              timeout: number;
+              excludeCredentials: {
+                  id: string;
+                  type: "public-key";
+                  transports: ("ble" | "hybrid" | "internal" | "nfc" | "usb")[];
+              }[];
+              attestation: "none" | "indirect" | "direct" | "enterprise";
+              pubKeyCredParams: {
+                  alg: number;
+                  type: "public-key";
+              }[];
+              authenticatorSelection: {
+                  requireResidentKey: boolean;
+                  residentKey: "required" | "preferred" | "discouraged";
+                  userVerification: "required" | "preferred" | "discouraged";
+              };
+          }
+        | { status: "RECOVER_ACCOUNT_TOKEN_INVALID_ERROR" }
+        | { status: "INVALID_EMAIL_ERROR"; err: string }
+    > {
+        let payload: { email: string } | { recoverAccountToken: string } | null = email
+            ? { email }
+            : recoverAccountToken
+            ? { recoverAccountToken }
+            : null;
+
+        if (!payload) {
+            return { status: "INVALID_EMAIL_ERROR", err: "Email is missing" };
+        }
+
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.registerOptions({
+            requireResidentKey: DEFAULT_REGISTER_OPTIONS_REQUIRE_RESIDENT_KEY,
+            residentKey: DEFAULT_REGISTER_OPTIONS_RESIDENT_KEY,
+            userVerification: DEFAULT_REGISTER_OPTIONS_USER_VERIFICATION,
+            supportedAlgorithmIds: DEFAULT_REGISTER_OPTIONS_SUPPORTED_ALGORITHM_IDS,
```

Comment:
These are overridable only through overrides (by overriding the methods). This should suffice, because the assumption is that only enterprise customers would need to override these. The regular customers would use our defaults.